silk/tests/openthread/ot_test_off_mesh_route_traffic.py

```diff
@@ -0,0 +1,233 @@
+        self.r1.add_ip6_address_on_interface(OFF_MESH_ADDR_1, prefix_len=64)
+        self.wait_for_completion(self.device_list)
+
+        self.r2.add_route_using_prefix(OFF_MESH_ROUTE_2)
+        self.r2.add_ip6_address_on_interface(OFF_MESH_ADDR_2, prefix_len=64)
+        self.wait_for_completion(self.device_list)
+
+        self.fed1.add_route_using_prefix(OFF_MESH_ROUTE_3)
+        self.fed1.add_ip6_address_on_interface(OFF_MESH_ADDR_3, prefix_len=64)
+        self.wait_for_completion(self.device_list)
+
+        # Wait till network data is updated on r1, r2, and sed2 and they all see all
+        # the added off-mesh routes.
+        time.sleep(WAIT_TIME)
+
+        def check_off_mesh_routes():
+            # If a node itself adds a route, the route entry will be seen twice in
+            # its WPAN_THREAD_OFF_MESH_ROUTES list (one time as part of network-wide
+            # network data and again as part of the local network data). Note that
+            # `r1 and `r2` each add a route, while `sed2` does not.
+            r1_routes = self.r1.wpanctl("get", "get " + wpan.WPAN_THREAD_OFF_MESH_ROUTES, 2)
+            r1_routes = wpan_table_parser.parse_list(r1_routes)
+            self.assertEqual(len(r1_routes), NUM_ROUTES + NUM_ROUTES_LOCAL)
+
+            r2_routes = self.r2.wpanctl("get", "get " + wpan.WPAN_THREAD_OFF_MESH_ROUTES, 2)
+            r2_routes = wpan_table_parser.parse_list(r2_routes)
+            self.assertEqual(len(r2_routes), NUM_ROUTES + NUM_ROUTES_LOCAL)
+
+            sed2_routes = self.sed2.wpanctl("get", "get " + wpan.WPAN_THREAD_OFF_MESH_ROUTES, 2)
+            sed2_routes = wpan_table_parser.parse_list(sed2_routes)
+            self.assertEqual(len(sed2_routes), NUM_ROUTES)
+
+        verify_within(check_off_mesh_routes, WAIT_TIME)
+
+    @testcase.test_method_decorator
+    def test04_transmit_receive(self):
+        # Traffic from `sed2` to `OFF_MESH_ADDR_1` (verify that it is received on`r1`).
+        # Traffic from `r1` to `OFF_MESH_ADDR_2` (verify that it is received on `r2`),
+        # Traffic from `r2` to `OFF_MESH_ADDR_3` (verify that it is received on `fed1`)
+        class AddressType(enum.Enum):
+            Prefix = 0
+
+        addresses = [
+            (self.sed2, self.r1, AddressType.Prefix, OFF_MESH_ADDR_1),
+            (self.r1, self.r2, AddressType.Prefix, OFF_MESH_ADDR_2),
+            (self.r2, self.fed1, AddressType.Prefix, OFF_MESH_ADDR_3),
+        ]
+
+        timeout = 5
+        delay = 1
+        for i, (src, dst, src_type, dst_address) in enumerate(addresses):
+            port = random.randint(10000 + i * 100, 10099 + i * 100)
+            message = random_string(10)
+            src_address = ""
+            if src_type == AddressType.Prefix:
+                address = src.find_ip6_address_with_prefix(ON_MESH_PREFIX)
+                self.wait_for_completion(self.device_list)
+                src_address = f"{address}%{src.netns}"
```

Comment:
I think we only need to attach src.netns for LLA address.

Reply:
Ack. This fixed the issue.